test/lit/passes/j2cl-merge-itables.wast

```diff
@@ -0,0 +1,218 @@
+    ;; CHECK:       (type $SubObject.vtable (sub $Object.vtable (struct (field structref) (field (ref $function)))))
+    (type $SubObject.vtable (sub $Object.vtable (struct (field (ref $function)))))
+
+    (type $SubObject.itable (sub $Object.itable
+        (struct (field (ref null struct)))))
+
+    (type $SubObject (sub $Object (struct
+        (field $vtable (ref $SubObject.vtable))
+        (field $itable (ref $SubObject.itable)))))
+
+    (type $function (func))
+  )
+
+  ;; Object vtable and itable initialization.
+  ;; CHECK:      (global $SubObject.vtable (ref $SubObject.vtable) (struct.new $SubObject.vtable
+  ;; CHECK-NEXT:  (ref.null none)
+  ;; CHECK-NEXT:  (ref.func $SubObject.f)
+  ;; CHECK-NEXT: ))
+
+  ;; CHECK:      (global $SubObject.itable (ref $SubObject.itable) (struct.new_default $SubObject.itable))
+
+  ;; CHECK:      (global $Object.vtable (ref $Object.vtable) (struct.new $Object.vtable
+  ;; CHECK-NEXT:  (ref.null none)
+  ;; CHECK-NEXT: ))
+  (global $Object.vtable (ref $Object.vtable)
+      (struct.new $Object.vtable))
+  ;; CHECK:      (global $Object.itable (ref $Object.itable) (struct.new_default $Object.itable))
+  (global $Object.itable (ref $Object.itable)
+      (struct.new_default $Object.itable))
+
+  ;; SubObject vtable and itable initialization. Shared empty itable.
+  (global $SubObject.vtable (ref $SubObject.vtable)
+      (struct.new $SubObject.vtable (ref.func $SubObject.f)))
+  (global $SubObject.itable (ref $SubObject.itable)
+      (struct.new_default $SubObject.itable))
+
+  ;; CHECK:      (func $SubObject.f (type $function)
+  ;; CHECK-NEXT:  (nop)
+  ;; CHECK-NEXT: )
+  (func $SubObject.f
+      (type $function)
+  )
+
+  ;; CHECK:      (func $usages (type $2)
+  ;; CHECK-NEXT:  (local $o (ref null $SubObject))
+  ;; CHECK-NEXT:  (local.set $o
+  ;; CHECK-NEXT:   (struct.new $SubObject
+  ;; CHECK-NEXT:    (global.get $SubObject.vtable)
+  ;; CHECK-NEXT:    (global.get $SubObject.itable)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (struct.get $SubObject.vtable 1
+  ;; CHECK-NEXT:    (struct.get $SubObject $vtable
+  ;; CHECK-NEXT:     (local.get $o)
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (struct.get $SubObject.vtable 0
+  ;; CHECK-NEXT:    (struct.get $SubObject $vtable
+  ;; CHECK-NEXT:     (local.get $o)
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $usages
+    (local $o (ref null $SubObject))
+    (local.set $o
+        (struct.new $SubObject
+              (global.get $SubObject.vtable)
+              (global.get $SubObject.itable)))
+    (drop
+        (struct.get $SubObject.vtable 0
+            (struct.get $SubObject $vtable
+                (local.get $o))))
+    (drop
+        (struct.get $Object.itable 0
+            (struct.get $SubObject $itable
+                (local.get $o))))
```

Comment:
Please indent these the way the other binaryen tests are formatted: two spaces. (We also use 1 in some places, for large tests, so that is an option too, but doesn't seem necessary here.)